libs/pageserver_api/src/models.rs

```diff
@@ -455,6 +455,25 @@ pub enum CompactionAlgorithm {
     Tiered,
 }
 
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    Serialize,
+    Deserialize,
+    strum_macros::FromRepr,
+    strum_macros::EnumString,
+    enum_map::Enum,
+)]
+#[strum(serialize_all = "kebab-case")]
+pub enum ImageCompressionAlgorithm {
+    ZstdLow,
+    Zstd,
```

Comment:
Any reason not to Just make this a `Zstd(u8)`, rather than encoding named high/low levels?  Presuming they'd encode to json as {"zstd": 3} or something like that, which is a little bit less handy than just a string, but not too bad.

I'd like to avoid a situation where in future we end up with a "ZstdVeryHigh" or "ZstdSlightlyLower" :-)

Reply:
> Any reason not to Just make this a Zstd(u8)

mostly just because this PR was the result of a 40 minute effort to export a subset from https://github.com/neondatabase/neon/pull/7879 , it's already in the TODOs at the top :). I agree that a scalar makes more sense here.

> Presuming they'd encode to json as {"zstd": 3} or something like that

yeah it would probably be something like `{kind = "Zstd", level= 4242 }`.

Reply:
using an rich enum now